hi-ml-cpath/src/health_cpath/models/deepmil.py

```diff
@@ -94,14 +97,43 @@ def __init__(self,
         self.aggregation_fn, self.num_pooling = pooling_params.get_pooling_layer(self.encoder.num_encoding)
         self.classifier_fn = self.get_classifier()
         self.activation_fn = self.get_activation()
-
         self.loss_fn = self.get_loss()
 
         # Metrics Objects
         self.train_metrics = self.get_metrics()
         self.val_metrics = self.get_metrics()
         self.test_metrics = self.get_metrics()
 
+    @staticmethod
+    def copy_weights(current_submodule: nn.Module, pretrained_submodule: nn.Module, submodule_name: str) -> None:
+        """Copy weights from pretrained submodule to current submodule.
+
+        :param current_submodule: Submodule to copy weights to.
+        :param pretrained_submodule: Submodule to copy weights from.
+        :param submodule_name: Name of the submodule.
+        """
+
+        for param, pretrained_param in zip(current_submodule.parameters(), pretrained_submodule.parameters()):
+            try:
+                param.data.copy_(pretrained_param.data)
+            except Exception as e:
+                raise ValueError(f"Failed to copy weights for {submodule_name} because of the following exception: {e}")
```

Comment:
I think we should test this method - with a couple of different submodules. I see a test below to test weight options but that is not testing the values, only the number of calls. Things I can think we want to assert: values are copied correctly, no mismatches in the value types. 
We could also check here that the number of parameters is the same and raise an error if not (I can see the possibility we try by mistake to initialize a model with weights coming from a checkpoint trained with a different version of the model)

Reply:
Agree, will add more tests.